### PR TITLE
Very basic multiplayer implementation

### PR DESCRIPTION
**Pull request by reco1I**
The Kotlin version was upgraded to 1.7, be sure of update the plugin in your IDE in order to test this.

### CODE REVIEW

File: src/ru/nsu/ccfit/zuev/osu/menu/ModMenu.java [hunk 1]
```diff
@@ -129,26 +192,48 @@ public void init() {
...
         addButton(offset, Config.getRES_HEIGHT() / 2 - button.getHeight() * 3, "selection-mod-easy", GameMod.MOD_EASY);
-        addButton(offset + offsetGrowth, Config.getRES_HEIGHT() / 2 - button.getHeight() * 3, "selection-mod-nofail", GameMod.MOD_NOFAIL);
-        addButton(offset + offsetGrowth * 2, Config.getRES_HEIGHT() / 2 - button.getHeight() * 3, "selection-mod-halftime", GameMod.MOD_HALFTIME);
-        addButton(offset + offsetGrowth * 3, Config.getRES_HEIGHT() / 2 - button.getHeight() * 3, "selection-mod-reallyeasy", GameMod.MOD_REALLYEASY);
+
+        // Used to define the X offset of each button according to its visibility
+        int factor = 1;
+
+        if (!Multiplayer.isMultiplayer)
+            addButton(offset + offsetGrowth * factor++, Config.getRES_HEIGHT() / 2 - button.getHeight() * 3, "selection-mod-nofail", GameMod.MOD_NOFAIL);
```
Comment (Rian8337) [inline src/ru/nsu/ccfit/zuev/osu/menu/ModMenu.java:200]: NF should be usable in multiplayer. In the current state of multiplayer, it should allow online score submission regardless of fail status.

In the future, when we support displaying failed scores (that did not recover in time), NF would also prevent the player from going to failed state.

Comment (reco1I) [reply src/ru/nsu/ccfit/zuev/osu/menu/ModMenu.java:200]: But what is the purpose of NF in multiplayer?

Comment (Rian8337) [reply src/ru/nsu/ccfit/zuev/osu/menu/ModMenu.java:200]: > In the current state of multiplayer, it should allow online score submission regardless of fail status.

> In the future, when we support displaying failed scores (that did not recover in time), NF would also prevent the player from going to failed state.

File: src/ru/nsu/ccfit/zuev/osu/menu/ModMenu.java [hunk 2]
```diff
@@ -62,35 +67,93 @@ public static ModMenu getInstance() {
...
-        setSongMenu(songMenu);
+    public void show(Scene scene, TrackInfo selectedTrack) {
+        parent = scene;
         setSelectedTrack(selectedTrack);
-        songMenu.scene.setChildScene(getScene(),
-                false, true, true);
-        InGameSettingMenu.getInstance().show();
+        scene.setChildScene(getScene(), false, true, true);
+
+        // TODO Custom mods support for multiplayer.
```
[the remark this answers is not in the document]
Comment (reco1I) [reply src/ru/nsu/ccfit/zuev/osu/menu/ModMenu.java:80]: Sorry, didn't specify it well.
That fragment contains the custom mods (speed modify, custom AR and custom FL delay) UI, that's why im avoiding to show it there and added the TODO.

File: build.gradle [hunk 3]
```diff
@@ -30,7 +30,7 @@ def getReleaseVersionCode() {
 }
 
 def getReleaseVersionName() {
-    def baseline = '1.6.8'
+    def baseline = '1.7.0'
```
[the remark this answers is not in the document]
Comment (reco1I) [reply build.gradle:33]: Isn't it cool ?

File: AndroidManifest.xml [hunk 4]
```diff
@@ -104,6 +104,20 @@
...
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+
+                <data
+                        android:scheme="https"
+                        android:host="multi.osudroid.moe"
+                />
```
[the remark this answers is not in the document]
Comment (reco1I) [reply AndroidManifest.xml]: I changed it to `https://odmp/[Room ID]/`

**Comment by Rian8337** [comment]
Note that CI fails here due to Gradle bug. It should be resolved soon, but I suppose we can wait.

**Pull request merged**